### ISSUE TITLE
bug on login

### ISSUE THREAD

**Issue by Addonman** [BullseiWoWAddons/BattleGroundEnemies#51]
Message: Interface/AddOns/BattleGroundEnemies/dbupgrades.lua line 22:
   attempt to index local 'destination' (a nil value)
Debug:
   [string "=(tail call)"]: ?
   [string "=(tail call)"]: ?
   [string "@Interface/AddOns/PersoLootRoll/Core/Addon.lua"]:597:
      Interface/AddOns/PersoLootRoll/Core/Addon.lua:596
   [string "=[C]"]: ?
   [string "@Interface/AddOns/BattleGroundEnemies/dbupgrades.lua"]:22:
      Interface/AddOns/BattleGroundEnemies/dbupgrades.lua:17
   [string "@Interface/AddOns/BattleGroundEnemies/dbupgrades.lua"]:75: UpgradeProfile()
   [string "@Interface/AddOns/BattleGroundEnemies/dbupgrades.lua"]:96: UpgradeProfiles()
   [string "@Interface/AddOns/BattleGroundEnemies/Main.lua"]:3171: ?()
   [string "@Interface/AddOns/BattleGroundEnemies/Main.lua"]:2468:
      Interface/AddOns/BattleGroundEnemies/Main.lua:2465
Locals:

**Comment by bullsei** [comment]
Hey, thanks for your report, will be fixed in next release/alpha

**Comment by Addonman** [comment]
now is this error poping up with latest aplpha
Message: Interface/AddOns/BattleGroundEnemies/dbupgrades.lua line 95:
   attempt to index field 'Text' (a nil value)
Debug:
   [string "=(tail call)"]: ?
   [string "=(tail call)"]: ?
   [string "@Interface/AddOns/PersoLootRoll/Core/Addon.lua"]:597:
      Interface/AddOns/PersoLootRoll/Core/Addon.lua:596
   [string "=[C]"]: ?
   [string "@Interface/AddOns/BattleGroundEnemies/dbupgrades.lua"]:95: UpgradeProfile()
   [string "@Interface/AddOns/BattleGroundEnemies/dbupgrades.lua"]:115: UpgradeProfiles()
   [string "@Interface/AddOns/BattleGroundEnemies/Main.lua"]:1309: ?()
   [string "@Interface/AddOns/BattleGroundEnemies/Main.lua"]:598:
      Interface/AddOns/BattleGroundEnemies/Main.lua:595

**Comment by bullsei** [comment]
hey, i tried to play around getting old savedvariables upgraded to new system but kinda impossible..., newest alpha gets rid of it by resettngi to default settings,  not the most elegant solution, but since many settings are global now its impossible to merge many into one. with the new editmode and more global options its much easier to setup, i hope thatss worth it

**Comment by Addonman** [comment]
well i installed new version and wiped my settings with it. I didn't have much of settings so not a problem with me. No errors on login as well .

**Comment by Thaodan** [comment]
I'm not sure if these are related but the error happens at some point shortly after logging when moving from world to PvP and back. Main symptom is that the enemy doesn't activate in RBGS (haven't tested other instances).
```
3x BattleGroundEnemies/Mainframe.lua:458: attempt to index field 'config' (a nil value)
[string "@BattleGroundEnemies/Mainframe.lua"]:458: in function `CheckEnableState'
[string "@BattleGroundEnemies/Main.lua"]:1269: in function `Enable'
[string "@BattleGroundEnemies/Main.lua"]:2258: in function `?'
[string "@BattleGroundEnemies/Main.lua"]:598: in function <BattleGroundEnemies/Main.lua:595>

Locals:
self = BGEEnemies {
 PlayerSources = <table> {
 }
 Players = <table> {
 }
 NumPlayers = 0
 CurrentPlayerOrder = <table> {
 }
 ActiveProfile = FontString {
 }
 PlayerCount = FontString {
 }
 NewPlayersDetails = <table> {
 }
 Counter = <table> {
 }
 InactivePlayerButtons = <table> {
 }
 PlayerType = "Enemies"
}
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to index field 'config' (a nil value)"
BattleGroundEnemies = BattleGroundEnemies {
 CombatLogevents = <table> {
 }
 DebugText = ""
 GeneralEvents = <table> {
 }
 AllyFaction = 0
 states = <table> {
 }
 UserButton = BattleGroundEnemiesAlliesframe8 {
 }
 EnemyFaction = 1
 Allies = BGEAllies {
 }
 Editmode = <table> {
 }
 RetailEvents = <table> {
 }
 BattleGroundDebuffs = <table> {
 }
 consts = <table> {
 }
 PendingUpdates = <table> {
 }
 ArenaIDToPlayerButton = <table> {
 }
 Counter = <table> {
 }
 UserFaction = "Horde"
 enabled = true
 UserDetails = <table> {
 }
 currentFocus = false
 Mixins = <table> {
 }
 ClassicEvents = <table> {
 }
 PlayerGUIDs = <table> {
 }
 options = <table> {
 }
 currentTarget = false
 db = <table> {
 }
 SearchedGUIDs = <table> {
 }
 IsRatedBG = false
 WrathEvents = <table> {
 }
 EditMode = <table> {
 }
 CustomEditModeMagnetismManager = <table> {
 }
 Enemies = BGEEnemies {
 }
 Testmode = <table> {
 }
 CurrentMapID = false
 ButtonModules = <table> {
 }
 specCache = <table> {
 }
}
```

**Comment by bullsei** [comment]
Okay, thanks for the report will look into it once I am back home... Wondering what I accidentally might have missed up 👀

**Comment by bullsei** [comment]
@Thaodan So i looked into it, so it seems like the error happend after you got into pvp zone (bg). the addon sets the .config on login event, but on a 0.2 second delay, so my guess it that it wasn't set since you basically went instantly into the bg after login or reload ui. can you confirm this is what happend?

Either way, i just uploaded a new alpha which implements a fix, it should help if my theory is correct

**Comment by Thaodan** [comment]
bullsei ***@***.***> writes:

> @Thaodan So i looked into it, so it seems like the error happend after
> you got into pvp zone (bg). the addon sets the .config on login event,
> but on a 0.2 second delay, so my guess it that it wasn't set since you
> basically went instantly into the bg after login or reload ui. can you
> confirm this is what happend?

Yes and no. I entered the bg, had no enemy frame and thought a reload might
help after joining.
Then I reloaded, got the error and still no enemy frame.

**Comment by bullsei** [comment]
okay, got it, seems like the issue probably has been in the addon for longer, definitley before 10.0.2.3
its fixed now in the latest alpha, also fixed player dying event handling and priroty icon transparency, should be working fine now

**Comment by bullsei** [comment]
All of the errors and problems reported here should be fixed in version 11.0.5.1. Will close this for now, if you notice any problems feel free to create a new issue

**Comment by Thaodan** [comment]
Thanks! Works again.
